### PR TITLE
feat: added a bunch of small improvements regarding types and such

### PR DESCRIPTION
**Pull request by V1RE**
- feat: added a bunch of small improvements regarding types and such
- feat: Added nice node docker image autocompletion

### CODE REVIEW

File: lib.ts [hunk 1]
```diff
@@ -124,78 +138,95 @@ export type JobOptions = {
  */
 export class Job {
   /**
-   * Do not used. The _uuid property is unstable and should be considered an internal implementation detail.
-   * @internal
+   * Do not use. The _uuid property is unstable and should be considered an internal implementation detail.
    */
-  readonly _uuid: string;
+  protected readonly _uuid = crypto.randomUUID();
```
Comment (grant0417) [inline lib.ts]: We actually need to grab this value elsewhere so it cant be `protected` at the moment

File: lib.ts [hunk 2]
```diff
@@ -124,78 +138,95 @@ export type JobOptions = {
...
-  constructor(public name: string) {
-    if (inJob()) {
-      try {
-        Deno.statSync(`/run/secrets/${name}`);
-      } catch (_e) {
-        throw new Error(
-          `Secret \`${name}\` is not available in this job, make sure it is specified in the job options.`,
-        );
-      }
+  constructor(name: string) {
```
Comment (grant0417) [inline lib.ts]: We also need to grab this name from the class so the name needs to be `public`

**Review by grant0417 — COMMENTED** [review]
Hey awesome pr! Two smallish comments, also if you are open to it we could see about adding you to our CLI repo before we open it if you want to see about contributing there.

**Comment by V1RE** [comment]
> Hey awesome pr! Two smallish comments, also if you are open to it we could see about adding you to our CLI repo before we open it if you want to see about contributing there.

Will fix them now! Would love to be added to the CLI repo!

File: lib.ts [hunk 3]
```diff
@@ -124,78 +138,95 @@ export type JobOptions = {
  */
 export class Job {
   /**
-   * Do not used. The _uuid property is unstable and should be considered an internal implementation detail.
-   * @internal
+   * @deprecated Do not use. The _uuid property is unstable and should be considered an internal implementation detail.
```
Comment (V1RE) [inline lib.ts:141]: Used [`@deprecated`](https://jsdoc.app/tags-deprecated.html) here so it's crossed out in autocomplete.
image

**Pull request merged**